python/GangaLHCb/__init__.py

```diff
@@ -81,17 +63,28 @@ def _store_root_version():
 
 
 def _store_dirac_environment():
-    from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
-    diracversion = _guess_version('LHCBDIRAC')
+    from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache
     platform = os.environ['CMTOPT']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['LHCbDiracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
     fname = os.path.join(fdir, diracversion)
     if not os.path.exists(fname) or not os.path.getsize(fname):
-        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
-        cmd = 'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
-        env = execute(cmd)  # grab the stdout text
-        env = eval(env)  # env is a string so convert it to a dict
+      cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
+      env = execute(cmd)
+      try:
+        env = eval(env)
+      except SyntaxError:
+        msg = 'LHCbDirac version {version} does not exist'.format(version=diracversion)
+        logger.error(msg)
+        raise OptionValueError()
+      try:
         write_env_cache(env, fname)
+        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
+      except (OSError, IOError, TypeError):
+        msg = 'Unable to store LHCbDirac environment'
+        logger.error(msg)
+        raise
```

Comment:
I'm not sure how that would help? The three there are the ones that could possibly be thrown. What does the GangaException do?